hivemind/dht/__init__.py

```diff
@@ -254,6 +263,24 @@ def add_validators(self, record_validators: Iterable[RecordValidatorBase]) -> No
     async def _add_validators(self, node: DHTNode, record_validators: Iterable[RecordValidatorBase]) -> None:
         node.protocol.record_validator.extend(record_validators)
 
+    @property
+    def peer_id(self) -> PeerID:
+        if self._peer_id is None:
+            self._peer_id = self.run_coroutine(DHT._get_peer_id)
+        return self._peer_id
+
+    async def _get_peer_id(self, node: DHTNode) -> PeerID:
+        return node.peer_id
```

Comment:
Can this be a static method?

Reply:
```suggestion
    @staticmethod
    async def _get_peer_id(dht: DHT, node: DHTNode) -> PeerID:
        return node.peer_id
```
Rewriting it as follows gets rid of the typing error